core/src/main/scripts/mutationalSignatures.py

```diff
@@ -0,0 +1,369 @@
+    # Find meta files in study
+    mut_file, study_id = find_mutations_data(study_path)
+    mut_file = f"{study_path}/{mut_file}"
+    return mut_file, study_id
+
+
+def decipher_genome(maf_df):
+    """Try to get the genome build from a MAF file
+    
+    :param maf_df: pandas object, pandas dataframe containing 'NCBI_Build' column
+    :return: str, genome build (GRCh**)
+    """
+    genome = list(maf_df["NCBI_Build"].unique())
+    if len(genome) != 1:
+        raise ValueError(f"There are either no or multiple genome builds present in your MAF file: \'{genome}\'")
+    try:
+        genome = int(genome[0])
+    except ValueError:
+        try:
+            genome = int(genome[0][-2:])
+        except ValueError:
+            genome = input("Can't decipher genome build. Please enter manually (37/38): ")
+            genome = int(genome)
+    genome = "GRCh" + str(genome)
+    return genome
+
+
+def pre_process_maf(mut_file, tmp_dir):
+    """Pre-process the cBioPortal mutations data file for matrix generation 
+    
+    :param mut_file: str, path to mutations file
+    :param tmp_dir: str, path to temp folder
+    :return: path to pre-processed MAF, genome build
+    """
+
+    # Create temporary directory and copy file(s)
+    try:
+        os.mkdir(tmp_dir)
+    except FileExistsError:
+        answer = input(f"\nDirectory \'{tmp_dir}\' already exists. Remove directory (including contents) "
+                       f"and continue script? (y/n): ")
+        if answer.lower() in ["y", "yes"]:
+            print(f"Removing {tmp_dir}...")
+            try:
+                shutil.rmtree(tmp_dir)
+            except NotADirectoryError:
+                answer = input(f"{tmp_dir} is not a directory. Still remove? (y/n): ")
+                if answer.lower() in ["y", "yes"]:
+                    os.remove(tmp_dir)
+                else:
+                    raise NotADirectoryError(f"Not a directory: '{tmp_dir}'")
+            os.mkdir(tmp_dir)
+        else:
+            raise FileExistsError(f"{tmp_dir} directory exists. Please give another temporary directory using the "
+                                  f"--temp-dir flag.")
+
+    # TODO block input this to avoid memory overflow?
+    # Read in maf and take relevant columns
+    maf = pd.read_csv(mut_file, sep="\t", comment="#")
+    # Get genome from MAF file
+    genome = decipher_genome(maf)
+
+    # TODO select which alternative allele?
+    alt_al = "Tumor_Seq_Allele2"
+    headers = ["Chromosome", "Start_Position", "End_Position", "Reference_Allele", alt_al, "Tumor_Sample_Barcode"]
+    maf = maf[headers]
+    # Remap maf file using important parts, as cbioportal MAF format does not conform to
+    #  GFC MAF file format (column "Consequence" is not placed on the correct location).
+    new_headers = ["NA" for i in range(16)]
+    for i, j in enumerate([4, 5, 6, 10, 12, 15]):
+        new_headers[j] = headers[i]
+    maf = maf.reindex(new_headers, axis=1)
```

Comment:
The filling of columns is done because sigprofilermatrixgenerator takes columns by index, not by header. So the empty columns are needed. Alternative allele is something we discussed to determine automatically, because sometimes Tumor_Seq_Allele1 contains the variant. This still has to be discussed/implemented.